### ISSUE TITLE
[BUG] can't write BigTiff file using ImageOutput

### ISSUE THREAD

**Issue by tvatooms** [OpenImageIO/oiio#3156]
**Describe the bug**
Hi , 
I tried to write a bigtiff file containing one floating channel (more than 4Go) using openimage 2.3.0.
I both tried using tiling and scanlines writing, but It doesn't work (stop at 3,99Go).

However, when I try to directly write the tiff using LibTiff and opening Tiff file using : 

**tif = TIFFOpen("file.tif", "w8");**

It does work (Tiff header has magical BigTiff value 43)

The issue may come from the fact that openimage tiffoutput.cpp doesn't  call TIFFOpen with option "w8" enabled : seems missing the "8" digit which enables BigTiff option.

**To Reproduce**

**Doesn't work** 

```
unsigned int sizeX = 40000;
unsigned int sizeY = 40000;
std::vector<float> pixels;
pixels.resize(sizeX);
const QString fileName = QString("temp/resScanline.tif");
std::unique_ptr<ImageOutput> out = ImageOutput::create(fileName.toStdString());

// Use Create mode for the first level.
ImageOutput::OpenMode appendmode = ImageOutput::Create;
ImageSpec spec(sizeX, sizeY, 1, TypeDesc::FLOAT);

spec.attribute("compression", "none");

if(!out->open(fileName.toStdString(), spec, appendmode)) {

	exit(0);
}
for(size_t y = 0; y < sizeY; ++y) {
	for(size_t offset = 0; offset < sizeX; ++offset) {
		pixels[offset] = 0;
	}

	if(!out->write_scanline(y, 0, TypeDesc::TypeFloat, pixels.data())) {

		exit(0);
	}
}
out->close();
```

**Works**

```
const size_t width = 40000;
const size_t height = 40000;
const double pixels_per_cm = 1;

float* data = new float[width * sizeof(float)];


int x, y;


for(x = 0; x < width; x++) {
	data[x] = 0;
}


TIFF* tif= TIFFOpen("temp/scanline.tif", "w8");
std::cout << "write";
TIFFSetField(tif, TIFFTAG_IMAGEWIDTH, width);
TIFFSetField(tif, TIFFTAG_IMAGELENGTH, height);
TIFFSetField(tif, TIFFTAG_BITSPERSAMPLE, 32);
TIFFSetField(tif, TIFFTAG_COMPRESSION, COMPRESSION_LZW);
TIFFSetField(tif, TIFFTAG_PHOTOMETRIC, PHOTOMETRIC_MINISBLACK);
// TIFFSetField(tif, TIFFTAG_FILLORDER, FILLORDER_MSB2LSB);
TIFFSetField(tif, TIFFTAG_SAMPLEFORMAT, SAMPLEFORMAT_IEEEFP);
TIFFSetField(tif, TIFFTAG_ORIENTATION, ORIENTATION_TOPLEFT);
TIFFSetField(tif, TIFFTAG_SAMPLESPERPIXEL, 1);
TIFFSetField(tif, TIFFTAG_ROWSPERSTRIP, 8);
TIFFSetField(tif, TIFFTAG_RESOLUTIONUNIT, RESUNIT_CENTIMETER);
TIFFSetField(tif, TIFFTAG_XRESOLUTION, pixels_per_cm);
TIFFSetField(tif, TIFFTAG_YRESOLUTION, pixels_per_cm);
TIFFSetField(tif, TIFFTAG_PLANARCONFIG, PLANARCONFIG_CONTIG);

tdata_t l_buffer;
unsigned long l_buffer_size = width * sizeof(float); // LINE buffer for float

l_buffer = _TIFFmalloc(l_buffer_size);

for(int row = 0; row < height; row++) {
	// std::cout << row << height;
	memcpy(l_buffer, data, l_buffer_size);

	int ret = TIFFWriteScanline(tif, l_buffer, row, 0);
	if(ret == -1) {
		TIFFClose(tif);
		return -1;
	}
}

_TIFFfree(l_buffer);

TIFFClose(tif);
delete[] data;
return 0;
}
```

**Expected behavior**
Should work as when using directly libtiff

**Evidence**
Do you have error messages? (please quote exactly) Screenshots? Example
command lines or scripts that reliably reproduce the error? If it only
happens with certain image files, can you attach the smallest image you
can make that reproduces the problem?

**Platform information:**
 - OIIO branch/version: 2.3.0
 - OS: Windows 10
 - C++ compiler: MSVC
 - Any non-default build flags when you build OIIO:

**IF YOU ALREADY HAVE A CODE FIX:** There is no need to file a separate issue,
please just go straight to making a [pull request](https://github.com/OpenImageIO/oiio/pulls).

**Comment by lgritz** [comment]
You're quite right. I've proposed a fix in #3158. It's not perfect, there are some trade-offs, as discussed in the PR. But I think it's the right approach.

**Comment by tvatooms** [comment]
Hi Larry , thank you for this answer. i'll test your fix.
